Sources/PackageGraph/PackageGraph+Loading.swift

```diff
@@ -222,24 +213,28 @@ private func createResolvedPackages(
         // Establish the manifest-declared package dependencies.
         packageBuilder.dependencies = package.manifest.dependenciesRequired(for: packageBuilder.productFilter)
             .compactMap { dependency in
-                // Use the package name to lookup the dependency. The package name will be present in packages with tools version >= 5.2.
-                if let dependencyName = dependency.explicitName, let resolvedPackage = packageMapByName[dependencyName] {
+                // Use the package name to lookup the dependency.
+                // The package name will be present in packages with tools version >= 5.2.
+                if let dependencyName = dependency.explicitName,
+                   let resolvedPackage = packageMapByName[dependencyName]
+                {
                     return resolvedPackage
                 }
 
                 // Otherwise, look it up by its identity.
                 let url = mirrors.effectiveURL(forURL: dependency.url)
-                let resolvedPackage = packageMapByIdentity[PackageReference.computeIdentity(packageURL: url)]
+                let resolvedPackage = packageMapByIdentity[PackageIdentity(url).computedName]
```

Comment:
lookup done by computedName instead of the identity itself intentional?

Reply:
Yes. It's a bit confusing in this context, but `PackageIdentity.computedName` is equivalent to the old / current `PackageReference.computeIdentity(packageURL:)` method.

Reply:
I made this clearer with c3fbf4e08ddd54a24fd24b0900702b3e747ad65a.